packages/typespec-go/test/cadlranch/special-headers/clientreqidgroup/custom_client.go

```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 )
 
-func NewClientRequestIdClient(options *azcore.ClientOptions) (*ClientRequestIdClient, error) {
+func NewClientRequestIDClient(options *azcore.ClientOptions) (*ClientRequestIDClient, error) {
```

Comment:
Ditto about API when it comes to `ID`. Is `ID` always capitilized in GO?

Reply:
Acronyms are always capitalized in Go and we have a list in the code generator that `ensureNameCase()` uses in order to perform the capitalization.